tests/integration_tests/performance/test_net_tput.py

```diff
@@ -0,0 +1,278 @@
+    test = 'tcp-{}-bw{}-ps{}-ws{}'.format(mode, bw, packet_size, 
+                                          WINDOW_SIZE_MB if window_size == 0 else '-')
+    total_sent = result['end']['sum_sent']
+    total_received = result['end']['sum_received']
+    duration = float(total_received['seconds'])
+    total_bytes_received = int(total_received['bytes'])
+    retransmits = int(total_sent['retransmits'])
+
+    # Measuring TPUT from rx-ing side.
+    tput = round((total_bytes_received*8) / (1024*1024*duration), 2)
+
+    if test not in tcp_statistics:
+        tcp_statistics[test] = create_tcp_statistics()
+
+    tcp_statistics[test]['tput'].add(tput)
+    tcp_statistics[test]['retransmits'].add(retransmits)
+
+    return {
+        'test': test,
+        'tput': tput,
+        'retransmits': retransmits,
+    }
+
+
+def run_iperf(microvm, host_ip, results, iperfs, protocol):
+    """Run tests using given protocol."""
+    run_statistics = {}
+
+    for _ in range(RUNS):
+        for bw in BANDWIDTHS[protocol]:
+            for mode in MODES.items():
+                for window_size in range(2):
+                    for packet_size in PACKET_SIZE_KB:
+                        cmd = '{} -c {} -t{} -f MBytes -J -b {}M {} {} -A {} \
+                                -P {} {} {} {}'.format(
+                            IPERF,
+                            host_ip,
+                            ITERATION_DURATION,
+                            bw,
+                            mode[1],
+                            PROTOCOL[protocol],
+                            PINNING_CPU,
+                            iperfs,
+                            '-w {}'.format(WINDOW_SIZE_MB) if protocol == 'tcp' and window_size == 0  else '',
+                            '-O {}'.format(OMIT_TIME) if protocol == 'tcp' else '',
+                            '-l {}'.format(packet_size) if protocol == 'tcp' else '',
+                        )
+
+                        # Don't do multiple runs on udp with the same parameter.
+                        if packet_size > PACKET_SIZE_KB[0] and protocol == 'udp':
+                            continue
+                        if window_size == 1 and protocol == 'udp':
+                            continue
+
+                        iperf_out = spawn_guest_iperf(microvm, cmd)
+                        if protocol == 'tcp':
+                            result = parse_iperf_tcp_result(iperf_out, mode[0],
+                                                            bw, window_size,
+                                                            packet_size, run_statistics)
+                        else:
+                            result = parse_iperf_udp_result(iperf_out, mode[0],
+                                                            bw, window_size, 
+                                                            run_statistics)
+                        results.append(result)
+                        print(result)
```

Comment:
I wouldn't add protocol-related logic here. I see `run_iperf` as a generic method that just runs the iperf client and takes care of the logic that's common to tcp and udp. 

We can have classes, for example IperfClient and IperfUdpClient/IperfTcpClient that extend IperfClient. Or we can just have methods. But I would separate the logic that's specific to Udp/Tcp from the common logic somehow. Since we also have different statistics for UDP and TCP, maybe it would make more sense to create different different classes for TCP and UDP that takes care both of the specific iperf client logic and also of the specific statistics logic.